app/services/api/create_trainee.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Api
+  class CreateTrainee
+    include ServicePattern
+
+    attr_accessor :current_provider, :trainee_attributes
+
+    def initialize(current_provider:, trainee_attributes:)
+      @current_provider = current_provider
+      @trainee_attributes = trainee_attributes
+    end
+
+    def call
+      unless trainee_attributes.valid?
```

Comment:
This is just a spike but in the real implementation we should method extract these otherwise Rubocop will start complaining but block length or something